### ISSUE TITLE
Docs: Filter example contains error

### ISSUE THREAD

**Issue by bocytko** [zalando/skipper#2280]
**Describe the bug**

Taking the example filter code from https://opensource.zalando.com/skipper/tutorials/development/#filters results in an error:

```
cannot use NewMyFilter() (value of type filters.Spec) as filters.Filter value in return statement: filters.Spec does not implement filters.Filter (missing method Request)
```

**To Reproduce**
Add context, for example:
1. Setup new filter using [example from docs](https://opensource.zalando.com/skipper/tutorials/development/#filters).

**Expected behavior**

```diff
@@ -11,7 +11,7 @@ func NewMyFilter() filters.Spec {
 func (spec *myFilter) Name() string { return "myFilter" }
 
 func (spec *myFilter) CreateFilter(config []interface{}) (filters.Filter, error) {
-       return NewMyFilter(), nil
+       return &myFilter{}, nil
 }
 
 func (f *myFilter) Request(ctx filters.FilterContext) {
```

There is another example under https://opensource.zalando.com/skipper/reference/plugins/#example-filter-plugin which is complete and compiles. Maybe worth deduplicating the documentation here.

**Observed behavior**
Compiler error when using example from documentation.

**Comment by szuecs** [comment]
I think a better fix should be

func NewMyFilter() *myFilter {
    return &myFilter{}
}

Maybe we need to expose the type by changing it to MyFilter, but this should work.
I guess somehow unboxing doesn't work as expected in case we have interface A implemented by T which also implements B, but as A can't be converted to B automatically.

**Comment by bocytko** [comment]
@szuecs you're the expert. I just provided an example that compiled and worked when tested.

**Comment by AlexanderYastrebov** [comment]
Fixed by #2425
